source/contracts/reporting/Market.sol

```diff
@@ -163,10 +164,13 @@ contract Market is DelegationTarget, ITyped, Initializable, Ownable, IMarket {
         if (IDisputeCrowdsourcer(_reportingParticipant).getSize() >= universe.getDisputeThresholdForFork()) {
             universe.fork();
         } else {
+            IFeeWindow _originalFeeWindow = feeWindow;
             feeWindow = universe.getOrCreateNextFeeWindow();
-            // Participants is implicitly bounded by the floor of the initial report REP cost to be no more than 21
-            for (uint256 i = 0; i < participants.length; i++) {
-                participants[i].migrate();
+            if (feeWindow != _originalFeeWindow) {
```

Comment:
BaseReportingParticipant.migrate() is a function which mints fee tokens for a reporting participant, rewarding those who have exposure over longer periods of time due to dispute process. Only when a market's feeWindow changes do we want to run this process, since less than a feeWindow's time has elapsed.
Additionally, since this logic would be getting their balance on a FeeWindow's token and doubling to it, migrate()ing every time would double their tokens every dispute round (instead of multiplying it by dispute round, across different FeeWindows), significantly rewarding early reporting participants.